### DISCUSSION TITLE
Without click it is running

### DISCUSSION THREAD

**Discussion by KurAgp** [flet-dev/flet#456] [General]
```
def btn_click():
    print('test')

def main(page:Page):
  
    page.add(FilledButton(text="Filled button",on_click=btn_click()))

    

flet.app(target=main)
```
This is simple button test program. When I run it with 'python button.py', it is print the 'test'. Expected behavior is when I click the button it will print. What can be the problem?

System: pop_os

**Comment by engamir1** [comment]
can u remove () in onclick

**Comment by FeodorFitsner** [comment]
it definitely should be `on_click=btn_click`, without `()`.
